src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/MicrosoftApiDesignGuidelinesAnalyzersResources.resx

```diff
@@ -514,7 +514,7 @@
     <value>To broaden the usability of a collection, implement one of the generic collection interfaces. Then the collection can be used to populate generic collection types.</value>
   </data>
   <data name="CollectionsShouldImplementGenericInterfaceMessage" xml:space="preserve">
-    <value>Collections should implement generic interface</value>
+    <value>Collection '{0}' directly or indirectly inherits '{1}' without implementing '{2}'. Publicly-visible collections should implement the generic version to broaden usability.</value>
```

Comment:
:bulb:

> Type '{0}' implements '{1}' without implementing '{2}'. Publicly-visible collection types should implement generic collection interfaces to improve usability.

Reply:
That's what I originally had it as. We decided to include the reference about indirect inheritance to remind people that that's a possibility.

Reply:
Note that I made small changes to several sections of the text. The current form seems too wordy; you can (and should) place additional information in the document pointed to by the Help Link URI.